--- a/Makefile
+++ b/Makefile
@@ -130,7 +130,7 @@ LIBBTRFS_LIBS = $(LIBS_BASE) $(LIBS_CRYPTO)
 
 # Static compilation flags
 STATIC_CFLAGS = $(CFLAGS) -ffunction-sections -fdata-sections -DSTATICBUILD=1
-STATIC_LDFLAGS = -static -Wl,--gc-sections
+STATIC_LDFLAGS = $(LDFLAGS) -static -Wl,--gc-sections
 STATIC_LIBS = $(STATIC_LIBS_BASE)
 
 # Static build with libmount print some warnings due to usage of UID/GID
 
