https://lists.gnupg.org/pipermail/gnupg-devel/2018-October/033985.html

--- a/m4/gpg-error.m4
+++ b/m4/gpg-error.m4
@@ -26,6 +26,7 @@ dnl is added to the gpg_config_script_warn variable.
 dnl
 AC_DEFUN([AM_PATH_GPG_ERROR],
 [ AC_REQUIRE([AC_CANONICAL_HOST])
+  AC_REQUIRE([PKG_PROG_PKG_CONFIG])
   gpg_error_config_prefix=""
   dnl --with-libgpg-error-prefix=PFX is the preferred name for this option,
   dnl since that is consistent with how our three siblings use the directory/
@@ -41,6 +42,23 @@ AC_DEFUN([AM_PATH_GPG_ERROR],
   AC_ARG_WITH(gpg-error-prefix,,
               [gpg_error_config_prefix="$withval"])
 
+  min_gpg_error_version=ifelse([$1], ,1.33,$1)
+
+  gpg_error_use_config=yes
+  if test x"${enable_pkg_config}" = xyes; then
+    PKG_CHECK_MODULES(
+      [GPG_ERROR],
+      [gpg-error >= $min_gpg_error_version],
+      [
+        gpg_error_use_config=no
+        PKG_CHECK_VAR([GPG_ERROR_MT_CFLAGS], [gpg-error], [mtcflags])
+        PKG_CHECK_VAR([GPG_ERROR_MT_LIBS], [gpg-error], [mtlibs])
+      ],
+      [:]
+    )
+  fi
+
+  if test x"${gpg_error_use_config}" = xyes; then
   if test x"${GPG_ERROR_CONFIG}" = x ; then
      if test x"${gpg_error_config_prefix}" != x ; then
         GPG_ERROR_CONFIG="${gpg_error_config_prefix}/bin/gpg-error-config"
@@ -61,7 +79,6 @@ AC_DEFUN([AM_PATH_GPG_ERROR],
   fi
 
   AC_PATH_PROG(GPG_ERROR_CONFIG, gpg-error-config, no)
-  min_gpg_error_version=ifelse([$1], ,1.33,$1)
   ok=no
 
   if test "$prefix" = NONE ; then
@@ -185,4 +202,5 @@ AC_DEFUN([AM_PATH_GPG_ERROR],
   AC_SUBST(GPG_ERROR_LIBS)
   AC_SUBST(GPG_ERROR_MT_CFLAGS)
   AC_SUBST(GPG_ERROR_MT_LIBS)
+  fi
 ])
 
