diff --git a/usr/Makefile b/usr/Makefile
index c1866b6..2990060 100644
--- a/usr/Makefile
+++ b/usr/Makefile
@@ -61,7 +61,7 @@ iscsiadm: $(ISCSI_LIB_SRCS) $(DISCOVERY_SRCS) iscsiadm.o session_mgmt.o mntcheck
 
 iscsistart: $(ISCSI_LIB_SRCS) $(INITIATOR_SRCS) $(FW_BOOT_SRCS) \
 		iscsistart.o statics.o
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ -o $@ -lrt
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ -o $@ -static -lrt
 clean:
 	rm -f *.o $(PROGRAMS) .depend $(LIBSYS)
 
